odl/solvers/functional/default_functionals.py

```diff
@@ -0,0 +1,683 @@
+            Element in the domain of the functional.
+
+        Returns
+        -------
+        `self(x)` : `element` in the `field` of the ``domain``.
+            Evaluation of the functional.
+        """
+        return np.abs(x).inner(self.domain.one())
+
+    @property
+    def gradient(x):
+        """Gradient operator of the L1-functional."""
+        # It does not exist on all of the space
+        raise NotImplementedError
+
+    def proximal(self, sigma=1.0):
+        """Return the proximal operator of the L1-functional.
+
+        Parameters
+        ----------
+        sigma : positive float, optional
+            Regularization parameter of the proximal operator
+
+        Returns
+        -------
+        out : Operator
+            Domain and range equal to domain of functional
+        """
+
+        return proximal_l1(space=self.domain)(sigma)
+
+    @property
+    def conjugate_functional(self):
+        """The convex conjugate functional of the L1-norm."""
+        functional = self
+
+        class L1ConjugateFunctional(Functional):
+            """The convex conjugate functional of the L1-norm."""
+            def __init__(self):
+                """Initialize a new instance of the L1ConjugateFunctional."""
+                super().__init__(functional.domain, linear=False)
+                self._orig_func = functional
+
+            def _call(self, x):
+                """Applies the convex conjugate functional of the L1-norm to
+                the given point.
+
+                Parameters
+                ----------
+                x : `LinearSpaceVector`
+                    Element in the domain of the functional.
+
+                Returns
+                -------
+                `self(x)` : `element` in the `field` of the ``domain``.
+                    Evaluation of the functional.
+                """
+                if np.max(np.abs(x)) > 1:
+                    return np.inf
+                else:
+                    return 0
+
+            @property
+            def gradient(x):
+                """Gradient operator of the conjugate functional of the
+                L1-norm.
+                """
+                # It does not exist on all of the space, only for ||x|| < 1
+                # where it is the ZeroOperator
+                raise NotImplementedError
+
+            @property
+            def conjugate_functional(self):
+                """The conjugate functional of the conjugate functional of the
+                L1-norm.
+
+                Notes
+                -----
+                Since the L1-norm is proper, convex and lower-semicontinuous,
+                by the Fenchel-Moreau theorem the convex conjugate functional
+                of the convex conjugate functional, also known as the
+                biconjugate, is the functional itself [BC2011]_.
+                """
+                return self._orig_func
+
+            def proximal(self, sigma=1.0):
```

Comment:
This syntax will not work with the solvers, they expect to get a proximal factory.